Dockerfile

```diff
@@ -1,14 +1,38 @@
 FROM python:3.8-buster
 
-RUN pip install --upgrade pip
+ARG THIS_USER="packpacker"
+ARG THIS_UID="1000"
+ARG THIS_GID="100"
+ENV HOME=/home/$THIS_USER
 
-ADD requirements.txt /tmp/requirements.txt
-RUN pip install -r /tmp/requirements.txt
+USER root
+
+ADD ./bin/fix-permissions /usr/local/bin/fix-permissions
+RUN chmod a+rx /usr/local/bin/fix-permissions
 
 # include user code
-ADD . /opt/code/packpack
-WORKDIR /opt/code/packpack
-RUN pip install -e .
+ENV CODE_DIR=/opt/code/packpack
+
+ADD . $CODE_DIR
+WORKDIR $CODE_DIR
+
+# lifted from https://github.com/jupyter/docker-stacks/blob/master/base-notebook/Dockerfile
+RUN echo "auth requisite pam_deny.so" >> /etc/pam.d/su && \
+    touch /etc/sudoers && \
+    sed -i.bak -e 's/^%admin/#%admin/' /etc/sudoers && \
+    sed -i.bak -e 's/^%sudo/#%sudo/' /etc/sudoers && \
+    useradd -m -s /bin/bash -N -u $THIS_UID $THIS_USER && \
+    chown $THIS_USER:$THIS_GID /opt/code/packpack && \
+    chmod g+w /etc/passwd
+RUN fix-permissions $HOME
+RUN fix-permissions "$(dirname $CODE_DIR)"
+
+USER $THIS_UID
+
+ENV PATH="/home/packpacker/.local/bin:$PATH"
+RUN pip install --user --upgrade pip
+RUN pip install --user -r requirements.txt
```

Comment:
we still use requirements.txt here because we want to install into the container's system packages, and not make a virtualenv like pipenv would